ovirt_imageio/client/_ui.py

```diff
@@ -58,26 +58,25 @@ def update(self, n):
         """
         with self.lock:
             self.done += n
-            now = self.now()
-            if now < self.next:
-                return
-
-            self.next = now + self.step
-            self._draw(now)
+            if self.size:
+                new_value = int(self.done / self.size * 100)
+                if new_value > self.value:
+                    self.value = new_value
+                    self._draw()
 
     def close(self):
         with self.lock:
             # If we wrote progress, we need to draw the last progress line.
             if self.done > 0:
-                self._draw(self.now(), last=True)
+                self._draw(last=True)
 
-    def _draw(self, now, last=False):
-        elapsed = now - self.start
+    def _draw(self, last=False):
+        elapsed = self.now() - self.start
 
         if self.size:
-            progress = "%6.2f%%" % (self.done / self.size * 100)
+            progress = "%3d%%" % self.value
```

Comment:
Need to normalize self.value to 0 in case size was specified when creating the progress, and add test.